arangod/Aql/Functions.cpp

```diff
@@ -944,18 +944,45 @@ AqlValue mergeParameters(ExpressionContext* expressionContext,
   if (initial.isArray() && n == 1) {
     // special case: a single array parameter
     // Create an empty document as start point
-    builder.openObject();
-    builder.close();
-    // merge in all other arguments
-    for (VPackSlice it : VPackArrayIterator(initialSlice)) {
-      if (!it.isObject()) {
-        registerInvalidArgumentWarning(expressionContext, funcName);
-        return AqlValue(AqlValueHintNull());
+    if (!recursive) {
+      // Fast path for large arrays
+      containers::FlatHashMap<std::string_view, VPackSlice> attributes;
+
+      // first we construct a map holding the latest value for a key
+      for (VPackSlice it : VPackArrayIterator(initialSlice)) {
+        if (!it.isObject()) {
+          registerInvalidArgumentWarning(expressionContext, funcName);
+          return AqlValue(AqlValueHintNull());
+        }
+        for (auto const& [key, value] : VPackObjectIterator(it)) {
```

Comment:
Could even be optimized to 
```
for (auto const& [key, value] : VPackObjectIterator(it, /*useSequentialIteration*/ true)) {
```
so it iterates the attributes in sequential order, and not in index table order. Iteration in sequential order is probably better for pretching data and will lead to less jumping around "randomly" in memory.